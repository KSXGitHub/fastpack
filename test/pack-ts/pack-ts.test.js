const { fpack } = require("../helpers.js");

process.chdir(__dirname);

test("ts-loader in development mode", async () => {
  const { bundle, stderr } = await fpack(
    ["index.ts", "--dev", "--preprocess=\\.ts$:ts-loader"],
    { copyOutputTo: "build-dev" }
  );
  expect(bundle).toMatchSnapshot();
});
