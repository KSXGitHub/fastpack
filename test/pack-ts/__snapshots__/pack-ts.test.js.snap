// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ts-loader in development mode 1`] = `
"(function(modules) {
  // The module cache
  var installedModules = {};

  // The require function
  function __fastpack_require__(moduleId) {

    // Check if module is in cache
    if(installedModules[moduleId]) {
      return installedModules[moduleId].exports;
    }
    // Create a new module (and put it into the cache)
    var module = installedModules[moduleId] = {
      id: moduleId,
      l: false,
      exports: {}
    };

    // Execute the module function
    modules[moduleId].call(
      module.exports,
      module,
      module.exports,
      __fastpack_require__,
      __fastpack_import__
    );

    // Flag the module as loaded
    module.l = true;

    // Return the exports of the module
    return module.exports;
  }

  function __fastpack_import__(moduleId) {
    return new Promise((resolve, reject) => {
      try {
        resolve(__fastpack_require__(moduleId));
      } catch (e) {
        reject(e);
      }
    });
  }

  // expose the modules object
  __fastpack_require__.m = modules;

  // expose the module cache
  __fastpack_require__.c = installedModules;

  return __fastpack_require__(__fastpack_require__.s = 'NM$$ts$$_$$loader$indexDOT$$js$$B$$indexDOT$$ts');
})
({
\\"NM$$ts$$_$$loader$indexDOT$$js$$B$$renderDOT$$ts\\": function(module, exports, __fastpack_require__, __fastpack_import__) {
\\"use strict\\";
function render() {
    document.body.innerHTML = \\"<h1> Hello World from TypeScript!</h1>\\";
}
module.exports = render;
},
\\"NM$$ts$$_$$loader$indexDOT$$js$$B$$indexDOT$$ts\\": function(module, exports, __fastpack_require__, __fastpack_import__) {
\\"use strict\\";
exports.__esModule = true;
var render = __fastpack_require__(/* \\"./render.ts\\" */ \\"NM$$ts$$_$$loader$indexDOT$$js$$B$$renderDOT$$ts\\");
render();

try {module.exports.__esModule = module.exports.__esModule || true}catch(_){}
},

});
"
`;

exports[`ts-loader in production mode 1`] = `
"(function() {

/* NM$$ts$$_$$loader$indexDOT$$js$$B$$renderDOT$$ts */

let $n__NM$$ts$$_$$loader$indexDOT$$js$$B$$renderDOT$$ts = { id: \\"NM$$ts$$_$$loader$indexDOT$$js$$B$$renderDOT$$ts\\", exports: {}};
\\"use strict\\";
function $i__NM$$ts$$_$$loader$indexDOT$$js$$B$$renderDOT$$ts__render() {
    document.body.innerHTML = \\"<h1> Hello World from TypeScript!</h1>\\";
}
$n__NM$$ts$$_$$loader$indexDOT$$js$$B$$renderDOT$$ts.exports = $i__NM$$ts$$_$$loader$indexDOT$$js$$B$$renderDOT$$ts__render;

try{$n__NM$$ts$$_$$loader$indexDOT$$js$$B$$renderDOT$$ts.exports.__esModule = $n__NM$$ts$$_$$loader$indexDOT$$js$$B$$renderDOT$$ts.exports.__esModule || false;}catch(_){}

/* NM$$ts$$_$$loader$indexDOT$$js$$B$$indexDOT$$ts */

let $n__NM$$ts$$_$$loader$indexDOT$$js$$B$$indexDOT$$ts = { id: \\"NM$$ts$$_$$loader$indexDOT$$js$$B$$indexDOT$$ts\\", exports: {}};
\\"use strict\\";
$n__NM$$ts$$_$$loader$indexDOT$$js$$B$$indexDOT$$ts.exports.__esModule = true;
var $i__NM$$ts$$_$$loader$indexDOT$$js$$B$$indexDOT$$ts__render = ($n__NM$$ts$$_$$loader$indexDOT$$js$$B$$renderDOT$$ts.exports);
$i__NM$$ts$$_$$loader$indexDOT$$js$$B$$indexDOT$$ts__render();

try{$n__NM$$ts$$_$$loader$indexDOT$$js$$B$$indexDOT$$ts.exports.__esModule = $n__NM$$ts$$_$$loader$indexDOT$$js$$B$$indexDOT$$ts.exports.__esModule || true;}catch(_){}
})()
"
`;
